src/Engine/Utils.java

```diff
@@ -501,6 +501,63 @@ public static void sortLocationsByTravelTime(Unit unit, ArrayList<XYCoord> mapLo
     TravelDistanceComparator tdc = new TravelDistanceComparator(unit, map);
     Collections.sort(mapLocations, tdc);
   }
+  
+
+  /** Returns a string describing the game statistics of a given CO */
+  public static String getGameStatusData(GameMap map, Commander viewed)
+  {
+    StringBuilder sb = new StringBuilder();
+
+    // map state
+    int income = 0;
+    int unitCount = 0;
+    int vehCount = 0;
+    int unitFunds = 0;
+    
+    for( int w = 0; w < map.mapWidth; ++w )
+    {
+      for( int h = 0; h < map.mapHeight; ++h )
+      {
+        Location loc = map.getLocation(w, h);
+        if( loc.isProfitable() && loc.getOwner() == viewed )
+        {
+          income += viewed.incomePerCity;
+        }
+        Unit resident = loc.getResident();
+        if( null != resident && resident.CO == viewed )
+        {
+          unitCount++;
+          if (resident.model.chassis != ChassisEnum.TROOP)
+            vehCount++;
+          unitFunds += resident.model.getCost() * resident.getHP() / resident.model.maxHP;
+        }
+      }
+    }
+
+    sb.append("Income:         ").append(income).append("\n");
+    sb.append("Unit count:     ").append(unitCount).append("\n");
+    sb.append("Vehicle count:  ").append(vehCount).append("\n");
+    sb.append("Unit funds:     ").append(unitFunds).append("\n");
+
+    // ability stats
+    double[] abilityCosts = viewed.getAbilityCosts();
+    if (abilityCosts.length > 0)
+    {
+      double abilityPower = viewed.getAbilityPower();
+      double untilNextPower = abilityCosts[abilityCosts.length - 1]; // init to our biggest cost, so we can only go down
```

Comment:
Are CO abilities guaranteed to be stored in order of cost?

Reply:
I was about to say "no, but they are for all current Commanders"... but that state of affairs can change in `master`.
Will fix.